README.md

```diff
@@ -60,20 +60,93 @@ Note: The only requirement for the ROCK is to have a `scripts/start-worker.sh`
 file, which will be used as the entry point for the charm to start the workload
 container.
 
-### Adding Environment Variables
+### Adding Secrets & Environment Variables
 
-The Temporal Worker operator can be used to inject environment variables that
-can be ingested by your workflows. This can be done using the Juju command line
-as follows:
+The Charmed Temporal Worker allows the user to configure multiple sources of
+environment variables and secrets to be injected into the workload container and
+consumed by the user's workflow definitions. These sources can be configured
+through the `secrets` config parameter of the charm. Below are the three sources
+of environment variables and secrets currently supported. A user may choose to
+use one, all or none of them. Once the `secrets.yaml` file is ready, it can be
+configured into the charm as follows:
 
 ```bash
-juju attach-resource temporal-worker-k8s env-file=path/to/.env
+juju config temporal-worker-k8s secrets=@/path/to/secrets.yaml
 ```
 
-#### **`.env`**
+These secrets can then be injested by the workflows by using the `os` package as
+follows:
 
+```python
+import os
+value1 = os.getenv("key1")
 ```
-VALUE=123
+
+#### Direct Environment Variables
+
+These are usually values that are not secret and can be stored as plaintext. An
+example is setting the application environment to `staging` or `production`.
+They can be set as follows:
+
+##### **`secrets.yaml`**
+
+```yaml
+secrets:
+  env:
+    - key1: value1
+    - key2: value2
+```
+
+#### Juju User Secrets (Requires Juju 3.3+)
+
+[Juju secrets](https://juju.is/docs/juju/manage-secrets) are values which can be
+stored in the model and accessed by the charm. To do so, you must first add the
+secret and grant the charm access to it:
+
+```bash
+juju add-secret my-secret key1=value1 key2=value2
+
+# Output: secret:<secret_id>
+
+juju grant-secret my-secret temporal-worker-k8s
+```
+
+The secrets can then be configured into the charm as follows:
+
+##### **`secrets.yaml`**
+
+```yaml
+secrets:
+  juju:
+    - secret-id: <secret_id>
+      key: key1
```

Comment:
does this mean that the name of the environment variable needs to match the name of the secret in juju or vault? There might be cases where that name is not known before hand.